### PR TITLE
Improve integration test failure reasons

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

None / #2224 

# Overview

- [x] Add helpers
    - `appendFailureReason :: String -> HUnitFailure -> HUnitFailure`
    - `counterexample :: String -> IO a -> IO a`
- [x] Make `verify` show the full response if a check fails
- [x] Make `eventually` use `appendFailureReason` instead of re-throwing using `fail`, to keep the error nicely rendered
- [x] I guess we should replace usage of `fail` with `expectationFailure`? 🤔  (yes, we definitely should!)

# Comments

### Previously
Skärmavbild 2020-10-13 kl  11 07 01

### Now
Skärmavbild 2020-10-13 kl  11 01 34

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Framework/DSL.hs [hunk 1]
```diff
@@ -1436,8 +1440,36 @@ wantedErrorButSuccess =
     fail . ("expected an error but got a successful response: " <>) . show
 
 -- | Apply 'a' to all actions in sequence
-verify :: (Monad m) => a -> [a -> m ()] -> m ()
-verify a = mapM_ (a &)
+verify :: Show a => a -> [a -> IO ()] -> IO ()
+verify a = counterexample msg . mapM_ (a &)
+  where
+    msg = "While verifying " ++ show a
```
Comment (Anviking) [inline lib/core-integration/src/Test/Integration/Framework/DSL.hs:1457]: I guess we could also show the index of the check… unless the `SrcLoc` already leads to it 🤔

Comment (Anviking) [reply lib/core-integration/src/Test/Integration/Framework/DSL.hs:1457]: Looking at the screenshots above, it seems like the source location _now_ does point to the exact item within the `verify` list.

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try-

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Review by piotr-iohk — APPROVED** [review]
LGTM.

**Review by paweljakubas — APPROVED** [review]
Nice! Improvements on more informative reporting when errors are always welcomed!

**Comment by Anviking** [comment]
bors try

**Pull request merged**